### PR TITLE
Added comments and an attribute to suscriber

### PR DESCRIPTION
**Pull request by wesdotcool**
The attribute_data in subscriber will allow users to put in arbitrary data in the subscriber.
Added comments in many places. More documentation to come.

### CODE REVIEW

File: lib/yesmail/subscriber.rb [hunk 1]
```diff
@@ -61,6 +72,7 @@ def get_user_id_from_email
       end
     end
 
+    # This will
```
Comment (jakeonrails) [inline lib/yesmail/subscriber.rb:75]: i'm dying from suspense.

File: .gitignore [hunk 2]
```diff
@@ -6,3 +6,4 @@ spec/*
 .rvmrc
 .rspec
 .idea/*
+*~
```
Comment (jakeonrails) [inline .gitignore:9]: haven't seen this one before, what is it for?

File: lib/yesmail/subscriber.rb (hunk 1 above)
Comment (wesdotcool) [inline lib/yesmail/subscriber.rb:75]: Oops. Not quite complete...

File: .gitignore (hunk 2 above)
Comment (wesdotcool) [inline .gitignore:9]: This ignores emacs autosave files, which are by default the file name with a tilde at the end, like `.gitignore~`

**Pull request merged**